src/torchmetrics/functional/shape/procrustes.py

```diff
@@ -0,0 +1,64 @@
+# Copyright The Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Tuple, Union
+
+import torch
+from torch import Tensor, linalg
+
+from torchmetrics.utilities.checks import _check_same_shape
+from torchmetrics.utilities.prints import rank_zero_warn
+
+
+def procrustes_disparity(
+    dataset1: Tensor, dataset2: Tensor, return_all: bool = False
+) -> Union[Tensor, Tuple[Tensor, Tensor, Tensor]]:
+    """Runs procrustrus analysis on a batch of data points.
+
+    Works similar ``scipy.spatial.procrustes`` but for batches of data points.
+
+    Args:
+        dataset1: The first set of data points
+        dataset2: The second set of data points
```

Comment:
I wouldn't name them dataset since this traditionally has a different meaning in deep learning. what about point_cloud or something?